chainer/links/model/vision/resnet.py

```diff
@@ -0,0 +1,447 @@
+            are not initialized by the pre-trained model, but the default
+            initializer used in the original paper.
+
+    Attributes:
+        available_layers (list of str): The list of available layer names
+            used by ``__call__`` and ``extract`` methods.
+
+    """
+
+    def __init__(self, pretrained_model='auto'):
+        if pretrained_model is not None:
+            # As a sampling process is time-consuming,
+            # we employ a zero initializer for faster computation.
+            kwargs = {'initialW': constant.Zero()}
+        else:
+            # employ default initializers used in the original paper
+            kwargs = {'initialW': None}
+        super(ResNet50Layers, self).__init__(
+            conv1=Convolution2D(3, 64, 7, 2, 3, **kwargs),
+            bn1=BatchNormalization(64),
+            res2=BuildingBlock(3, 64, 64, 256, 1, **kwargs),
+            res3=BuildingBlock(4, 256, 128, 512, 2, **kwargs),
+            res4=BuildingBlock(6, 512, 256, 1024, 2, **kwargs),
+            res5=BuildingBlock(3, 1024, 512, 2048, 2, **kwargs),
+            fc6=Linear(2048, 1000),
+        )
+        if pretrained_model == 'auto':
+            _retrieve(
+                'ResNet-50-model.npz', 'ResNet-50-model.caffemodel', self)
+        elif pretrained_model is not None:
+            npz.load_npz(pretrained_model, self)
+        self.functions = OrderedDict([
+            ('conv1', [self.conv1, self.bn1, relu]),
+            ('pool1', [lambda x: max_pooling_2d(x, ksize=3, stride=2)]),
+            ('res2', [self.res2]),
+            ('res3', [self.res3]),
+            ('res4', [self.res4]),
+            ('res5', [self.res5]),
+            ('pool5', [_global_average_pooling_2d]),
+            ('fc6', [self.fc6]),
+            ('prob', [softmax]),
+        ])
+
+    @property
+    def available_layers(self):
+        return list(self.functions.keys())
+
+    @classmethod
+    def convert_caffemodel_to_npz(cls, path_caffemodel, path_npz):
+        """Converts a pre-trained caffemodel to a chainer model.
+
+        Args:
+            path_caffemodel (str): Path of the pre-trained caffemodel.
+            path_npz (str): Path of the converted chainer model.
+        """
+
+        # As CaffeFunction uses shortcut symbols,
+        # we import CaffeFunction here.
+        from chainer.links.caffe.caffe_function import CaffeFunction
+        caffemodel = CaffeFunction(path_caffemodel)
+        chainermodel = cls(pretrained_model=None)
+        _transfer_resnet50(caffemodel, chainermodel)
+        npz.save_npz(path_npz, chainermodel, compression=False)
+
+    def __call__(self, x, layers=['prob'], test=True):
+        """Computes all the feature maps specified by ``layers``.
+
+        Args:
+            x (~chainer.Variable): Input variable.
+            layers (list of str): The list of layernames you want to extract.
+            test (bool): If ``True``, it runs in test mode.
```

Comment:
The `test` flag only concerns the BatchNormalization layers. Maybe write "If ``True``, BatchNormalization runs in test mode."